### PR TITLE
service/header: add network name to protocol ID

### LINKED ISSUE #493: service/header: exchangeProtocolID should include network name from `params`
Currently, we have a static protocol ID for the HeaderEx protocol, meaning that peers from different networks(devnet, testnet, mainnet) can speak to each other and exchange invalid headers by the same heights. Instead, the protocol ID of HeaderEx(service/header/p2p_exchange.go:19) should contain the network the node is on(we do not allow a node to be on multiple networks at the same time).

### PR DESCRIPTION
**Pull request by toanngosy**
This pull request would close #493.

### CODE REVIEW

**Review by renaynay — APPROVED** [review]
LGTM and thanks for the contribution @toanngosy !

@Wondertan is this the format you were thinking?

We also need to add a changelog entry for this.

**Comment by toanngosy** [comment]
Thanks @renaynay ! I added the changelog into `CHANGELOG-PENDING.md`.

**Review by renaynay — APPROVED** [review]

File: service/header/p2p_exchange.go [hunk 1]
```diff
@@ -13,10 +13,11 @@ import (
 
 	"github.com/celestiaorg/go-libp2p-messenger/serde"
 
+	params "github.com/celestiaorg/celestia-node/params"
 	pb "github.com/celestiaorg/celestia-node/service/header/pb"
 )
 
-var exchangeProtocolID = protocol.ID("/header-ex/v0.0.1")
+var exchangeProtocolID = protocol.ID(fmt.Sprintf("/header-ex-%s/v0.0.1", params.GetNetwork()))
```
Comment (Wondertan) [inline service/header/p2p_exchange.go:20]: I believe the format should be like `/header-ex/v0.0.01/network`. Sorry for not mentioning this in the issue

**Review by Wondertan — CHANGES_REQUESTED** [review]
Thank you for tackling this.

**Comment by toanngosy** [comment]
@Wondertan it should work now 😄

**Comment by Wondertan** [comment]
Pls look at the commit history, @toanngosy. There are multiple duplicates from the master. Seems like artifacts from the rebasr. Needs to be fixed

**Comment by toanngosy** [comment]
sorry, I fixed it. It is now up to date with the main. @Wondertan

**Review by Wondertan — APPROVED** [review]
:shipit:

**Review by renaynay — APPROVED** [review]
Thank you so much for your contribution @toanngosy ❤️

**Pull request merged**